### ISSUE TITLE
Error unmarshalling an error message to serviceError2

### ISSUE THREAD

**Issue by mikhailshilkov** [Azure/go-autorest#608]
While creating an Azure SQL Database, I get an error (because I specified some properties incorrectly) but the library fails with `json: cannot unmarshal array into Go struct field serviceError2.details of type map[string]interface {}` which is not helpful at all.

My call that fails:

```go
err = autorest.Respond(
		resp,
		k.client.ByInspecting(),
		azure.WithErrorUnlessStatusCode(http.StatusOK, http.StatusCreated),
		autorest.ByUnmarshallingJSON(&outputs),
		autorest.ByClosing())
```

I logged the body that I get:

```
{"code":"40645","message":"Servername \"someServer20210108\" cannot be empty or null. It can only be made up 
of lowercase letters \u0027a\u0027-\u0027z\u0027, the numbers 0-9 and the hyphen. The hyphen may not lead or 
trail in the name.","target":null,"details":[{"code":"40645","message":"Servername \"someServer20210108\" 
cannot be empty or null. It can only be made up of lowercase letters \u0027a\u0027-\u0027z\u0027, the 
numbers 0-9 and the hyphen. The hyphen may not lead or trail in the name.","target":null,"severity":"16"}],
"innererror":[]}
```

Does the service behave incorrectly here? Could `go-autorest` fail gracefully and maybe return an error which would contain the raw response body?

**Comment by jhendrixMSFT** [comment]
The problem here is `innererror` being an array.  OData v4 spec says it should be a single object.

I have a fix for this, both to handle discrepancies in the error format as well as including the raw JSON in the error message for an unmarshalling failure.

**Comment by jhendrixMSFT** [comment]
Fixed in `autorest/v0.11.17`
